packages/js/src/components/contentAnalysis/SeoAnalysis.js

```diff
@@ -189,6 +191,23 @@ class SeoAnalysis extends Component {
 		];
 	}
 
+	/**
+	 * Renders the AI Assessment Fixes button.
+	 *
+	 * @param {boolean} hasAIFixes Whether the assessment has AI fixes or not.
+	 * @param {string} id The assessment ID for which the AI fixes should be applied to.
+	 *
+	 * @returns {JSX.Element} The AI Assessment Fixes button.
+	 */
+	renderAIFixesButton = ( hasAIFixes, id ) => {
+		const isPremium = getL10nObject().isPremium;
+
+		// The reason of adding the check if Elementor is active or not is because `isBlockEditor` method also returns `true` for Elementor.
+		return hasAIFixes && isBlockEditor() && ! this.props.isElementor && (
```

Comment:
This helper to check if Block editor is active `isBlockEditor` helper `packages/js/src/helpers/isBlockEditor.js` is not quite accurate: it also returns `true` inside Elementor. We got the value from `WP_Screen::get()->is_block_editor();`. As a workaround to make sure that the AI button is not shown in Elementor, here I added an extra check that the editor is NOT Elementor. So far I haven't found a more reliable method to check for Block editor. If such method exist, I guess that method would be preferable here

Reply:
Hmm, it's not optimal -- I guess we could also consider removing https://github.com/Yoast/wordpress-seo/blob/97f4aa23c2820fb6e55d51cd9c60117e59296a99/src/integrations/third-party/elementor.php#L466 

But I'm OK with the current implementation as well.